lib/std/zig/CrossTarget.zig

```diff
@@ -231,6 +231,32 @@ pub const ParseOptions = struct {
     };
 };
 
+/// Parses versions with omitted patch components, such as "1.0",
+/// which SemanticVersion.parse is not capable of.
+fn parseVersion(string: []const u8) !SemanticVersion {
+    const parseVersionComponent = struct {
+        fn parseVersionComponent(component: []const u8) !usize {
+            return std.fmt.parseUnsigned(usize, component, 10) catch |err| {
+                switch (err) {
+                    error.InvalidCharacter => return error.InvalidVersion,
+                    error.Overflow => return error.Overflow,
+                }
+            };
+        }
+    }.parseVersionComponent;
+    var version_components = mem.split(u8, string, ".");
+    const major = version_components.first();
+    if (major.len == 0) return error.InvalidVersion;
+    const minor = version_components.next() orelse return error.InvalidVersion;
+    const patch = version_components.next() orelse "0";
```

Comment:
This version doesn't check that there aren't additional components after the patch. Same error is repeated in `zig/system/darwin/macos.zig`, would it instead be better to provide a properly tested non-standard function somewhere so that other people don't repeat it by implementing their own version?

Reply:
The above brings me to this comment: where shall we put this function and its tests? It should probably be an entirely internal function. `lib/std/zig/CrossTarget.zig` and `lib/std/zig/system/darwin/macos.zig` share `lib/std/zig/` so should it be `/lib/std/zig/version.zig` or something?

Reply:
I'd either put it in `SemanticVersion.zig` and clearly document it as non-standard or just `lib/std/zig.zig`.

Reply:
Hmm I was just doing this but now I'm thinking is `std.zig` really the right space for this? Version parsing like that isn't really part of parsing Zig code. Doesn't really seem to fit in that file.

Well, I was gonna say another motivation of this PR was to reduce the public API surface a bit, and really, after this missing check at the end it seems to be complete now. The code really isn't that complicated.
The advantage is that now we have a single, standard way to parse a version.
For all other cases the user has to write this themselves, and it really isn't that complicated, and also only an optional minor component seems kind of specific. Maybe they want more components to be optional, etc.
So this way it's also very formidable for the use case at hand.

So for now I went back to the way it was and fixed up the parsing in both places.
However if you can think of a better place, where ideally it doesn't get into the public std, maybe we can do it the way your proposing.

Maybe `lib/std/zig/version.zig`? Can we put it there publicly and then only use it in those two places and not expose it to the public std API surface?